src/classes/AFFL_MultiRecordType_TDTM.cls

```diff
@@ -136,21 +130,26 @@ public class AFFL_MultiRecordType_TDTM extends TDTM_Runnable {
 		            }
 		    	 }
 	    	 }
-	    	 afflMultiHasRunBefore = true;
+           TDTM_ProcessControl.setRecursionFlag(TDTM_ProcessControl.registeredTrigger.AFFL_MultiRecordType_TDTM_Before, true);
         }
 
         //AFTER INSERT
-        if(!afflMultiHasRunAfter && triggerAction == TDTM_Runnable.Action.AfterInsert) {
+        if(!TDTM_ProcessControl.getRecursionFlag(TDTM_ProcessControl.registeredTrigger.AFFL_MultiRecordType_TDTM_After) && triggerAction == TDTM_Runnable.Action.AfterInsert) {
 			 if(newlist != null && newlist.size() > 0) {
 		    	 for (SObject so : newlist) {
 		            Affiliation__c affl = (Affiliation__c)so;
 					createProgramEnrollmentIfNecessary(affl, dmlWrapper);
 		    	 }
 			 }
-			 afflMultiHasRunAfter = true;
+           TDTM_ProcessControl.setRecursionFlag(TDTM_ProcessControl.registeredTrigger.AFFL_MultiRecordType_TDTM_After, true);
 		}
+		TDTM_TriggerHandler.processDML(dmlWrapper, true);
+        dmlWrapper = null;
+        if (triggerAction == TDTM_Runnable.Action.AfterInsert){
+            TDTM_ProcessControl.setRecursionFlag(TDTM_ProcessControl.registeredTrigger.AFFL_MultiRecordType_TDTM_After, false);
```

Comment:
I think AFFL_MultiRecordType_TDTM_Before needs to be set back to false here too.